src/pages/Jobs.js

```diff
@@ -2,27 +2,52 @@ import React, { useState, useEffect } from "react";
 import JobList from "../components/JobCard";
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { Row, Col, Container } from "react-bootstrap";
+import { Row, Col, Container, Form } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+
 export default function Jobs() {
   let originalList = [];
-  let filteredList = [];
   let [jobs, setJobs] = useState([]);
-  let keyword = "";
+  let [keyword, setKeyword] = useState("");
+  const [encode, setEncode] = useState("");
+  let history = useHistory();
+  let searchTerm = [];
 
   const getData = async () => {
     //let url ="http://localhost:3001/jobs";
     let url = `https://my-json-server.typicode.com/legobitna/Itviec/jobs`;
     let data = await fetch(url);
     let result = await data.json();
     originalList = result;
+    getQuery(originalList);
+  };
 
-    setJobs(result);
+  const getQuery = (arr) => {
+    let filteredList = [];
+    let queryString = window.location.search;
+    let splitQueryString = queryString.split("?q=");
+    if (splitQueryString.length > 1) {
+      searchTerm = splitQueryString[1].split("%20");
+      filteredList = arr.filter((item) =>
+        searchTerm.some((key) =>
+          item.title.toLowerCase().includes(key.toLowerCase())
+        )
+      );
+    } else {
+      filteredList = arr;
+    }
+    setJobs(filteredList);
   };
+
   useEffect(() => {
     getData();
-  }, []);
+  }, [encode]);
```

Comment:
I don't think you need `encode`. 

This logic isn't right, because `getData` triggers an API request. Below, `search` shouldn't make a new API request. You only want to `getData` once, which is when the page loads. 

What you want to do is instead:

```
useEffect( () => {
   getQuery();
}, [search])
```

Reply:
And then, on load:


```
useEffect( () => {
   getData();
}, []);
```

Reply:
i made a 2 different useEffect like what you said 
and instead of encode, i replace this to search function 
but i got into infinite loop problem

Reply:
```
  const search = (e) => {
    e.preventDefault();
    setEncode(encodeURIComponent(keyword));
    history.push("/jobs/?q=" + encodeURIComponent(keyword));
  };
  
  useEffect(() => {
    getData();
  }, []);

  useEffect(() => {
    getQuery();
  }, [search]);
 ``` 
this is my code

Reply:
when i run it, it keep calling getData and getQuery
can you help me?

Reply:
hmmm

Reply:
if i change that [search] back to [encode] it stop infinite loop but the second use effect is not  called after first useEffect run, so i can't get any originalList at getQuery function so can't search

Reply:
```
const getData = async () => {
    console.log("here1")
    //let url ="http://localhost:3001/jobs";
    let url = `https://my-json-server.typicode.com/legobitna/Itviec/jobs`;
    let data = await fetch(url);
    let result = await data.json();
    originalList = result;
    setJobs(originalList);
    // getQuery(originalList);
    //getQuery();
  };

  const getQuery = () => {
    console.log("ori",originalList)
    console.log("here2")
    filteredList = originalList;
    let searchParam = query.get("q");
    if (searchParam) {
      searchTerm = searchParam.split(" ");
      filteredList = filteredList.filter(item =>
        searchTerm.some(key =>
          item.title.toLowerCase().includes(key.toLowerCase())
        )
      );
    } 
    setJobs(filteredList);
  };
```

Reply:
these are my getdata and getQuery source code as well

Reply:
Okay so i made new Commit i seperate 2 useEffect so the API will only call once when we load the page but but the function [search] in the useEffect is not work. when i google about it, ppl put state, not the function so my idea is we need encode. if you have better idea let me know!